### PR TITLE
UAR-299: Date of update statement

### PR DESCRIPTION
**Pull request by jkayyala**
### JIRA link
https://companieshouse.atlassian.net/browse/UAR-299

### Change description
Add changes for showing Date of the Update statement

### Work checklist

- [ ] Tests added where applicable
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

**Review by azafarCH — APPROVED** [review]

File: src/controllers/update/update.filing.date.controller.ts [hunk 1]
```diff
@@ -4,19 +4,24 @@ import { Session } from "@companieshouse/node-session-handler";
...
-import { OverseasEntityKey, Transactionkey } from '../../model/data.types.model';
+import { OverseasEntityKey, Transactionkey, InputDateKeys } from '../../model/data.types.model';
+import { FilingDateKey, FilingDateKeys } from '../../model/date.model';
 import { ApplicationData } from "../../model/application.model";
 
 export const get = (req: Request, res: Response, next: NextFunction) => {
   try {
     logger.debugRequest(req, `${req.method} ${req.route.path}`);
+    const appData = getApplicationData(req.session);
+    const filingDate = (appData.update?.[FilingDateKey]) ? mapDataObjectToFields(appData.update[FilingDateKey], FilingDateKeys, InputDateKeys) : {};
```
Comment (AndreKramer7654) [inline src/controllers/update/update.filing.date.controller.ts:18]: no need for first () but if you think it's more readable

File: src/controllers/update/update.filing.date.controller.ts [hunk 2]
```diff
@@ -38,6 +43,10 @@ export const post = async(req: Request, res: Response, next: NextFunction) => {
         appData[OverseasEntityKey] = await createOverseasEntity(req, session, transactionID, true);
         setExtraData(session, appData);
```
Comment (AndreKramer7654) [inline src/controllers/update/update.filing.date.controller.ts:44]: Do we remove this setExtraData as doing it on line 49?

File: test/controllers/update/update.filing.date.controller.spec.ts [hunk 3]
```diff
@@ -136,11 +164,27 @@ describe("Update Filing Date controller", () => {
       expect(resp.text).toContain(`${FOUND_REDIRECT_TO} ${config.OVERSEAS_ENTITY_PRESENTER_URL}`);
     });
 
+    test(`renders the ${config.UPDATE_FILING_DATE_PAGE} page with  error when filing date is empty`, async () => {
```
Comment (AndreKramer7654) [inline test/controllers/update/update.filing.date.controller.spec.ts:167]: extra space "with  error"

**Review by AndreKramer7654 — APPROVED** [review]

File: src/validation/fields/date.validation.ts [hunk 4]
```diff
@@ -80,6 +80,17 @@ export const identity_check_date_validations = [
...
 
+export const filing_date_validations = [
+  body("filing_date-day")
+    .custom((value, { req }) => checkDateFieldDay(req.body["filing_date-day"], req.body["filing_date-month"], req.body["filing_date-year"])),
+  body("filing_date-month")
+    .custom((value, { req }) => checkDateFieldMonth(ErrorMessages.MONTH, ErrorMessages.MONTH_AND_YEAR, req.body["filing_date-day"], req.body["filing_date-month"], req.body["filing_date-year"])),
+  body("filing_date-year")
+    .custom((value, { req }) => checkDateFieldYear(ErrorMessages.YEAR, ErrorMessages.YEAR_LENGTH, req.body["filing_date-day"], req.body["filing_date-month"], req.body["filing_date-year"])),
+  body("filing_date-day")
+    .custom((value, { req }) => checkDate(req.body["filing_date-day"], req.body["filing_date-month"], req.body["filing_date-year"])),
```
Comment (AndreKramer7654) [inline src/validation/fields/date.validation.ts:91]: We need another validation:

 checkDateIsInPastOrToday()

Comment (jkayyala) [reply src/validation/fields/date.validation.ts:91]: checkDate calls checkAllDateFields which calls checkDateIsInPastOrToday

Comment (AndreKramer7654) [reply src/validation/fields/date.validation.ts:91]: Good, all fine

**Pull request merged**